### ISSUE TITLE
bug(stepper): aria-disabled not properly updated for step header buttons in linear steppers

### ISSUE THREAD

**Issue by amysorto** [angular/components#23521]
#### Reproduction

Example of a linear stepper with this issue: https://stackblitz.com/angular/vbkvxdpeoml?file=src%2Fapp%2Fstepper-overview-example.ts

Steps to reproduce:
1. Enable linear mode
2. Fill out first step
3. Go back to step header using keyboard navigation
4. Move focus to next step 
 
#### Expected Behavior

The screen reader should read aloud the content of the button but not mention anything about being disabled since clicking the button will work.

#### Actual Behavior

The screen reader says the button is disabled.

This is because aria-disabled attribute is still set on the button, but shouldn't be anymore since the step is now complete.

#### Environment

- Angular: 12.2.4
- CDK/Material: 12.2.4
- Tested on Chrome with NVDA, ChromeVox, and VoiceOver.

**Comment by zarend** [comment]
Hi @amysorto ,

Thank you for reporting this issue. I an unable to repro on the provided stackblitz: https://stackblitz.com/angular/vbkvxdpeoml?file=src%2Fapp%2Fstepper-overview-example.ts. Below is a video capture of me trying to repro. Could you please let me know if I'm doing the correct reproduction steps? That way we can know if there is a bug that we need to fix.

https://user-images.githubusercontent.com/7720245/213253311-503c2744-ace9-449e-b714-13b2680779d5.mov

-Zach

**Comment by zarend** [comment]
Okay, I can reproduce now. I forgot to enable linear mode 🤦 . Bug can be reproduced without using a screenreader:

1. navigate to _Stepper Overview_ example on https://material.angular.io/components/stepper/overview
2. (_Fill out your address_ has gray disabled appearance and cannot be clicked on)
3. Type _asdf_ into name field
4. (_Fill out your address_ has gray disabled appearance but *can* be clicked on. It has `aria-disabled="true"` in the DOM.)

Seems like an invalid state for something to have disabled appearance and aria-disabled="true", but that the user is still able to interact with it.
